tests/test_err.py

```diff
@@ -30,3 +29,24 @@ def test_bogus_band_error():
 def test_cplerror_str():
     err = CPLE_BaseError(1, 1, "test123")
     assert str(err) == "test123"
+
+
+def test_issue2353(caplog, path_rgb_byte_tif):
+    """Ensure transformer doesn't leave errors behind."""
+    from rasterio.warp import calculate_default_transform
+
+    with rasterio.open(path_rgb_byte_tif) as src:
```

Comment:
This test doesn't really need the URL you provided @vincentsarago , but it does need a dataset. This test fails with the code in master.

Reply:
here is one dataset if you want https://rio-tiler-dev.s3.amazonaws.com/data/goes.tif

Reply:
Thanks, but we don't need a real dataset to reproduce the bug, we only need a CRS, height/width, and extent. I guess I could add a second test for the GOES file, though.